### PR TITLE
 Add allowEmptySession flag to allow not creating empty sessions

### PR DESCRIPTION
**Pull request by yotammadem**

### CODE REVIEW

**Review by jsumners — COMMENTED** [review]
1. The `package-lock.json` file will need to be removed
2. Being able to set an empty session is the mechanism by which sessions are invalidated. How would one invalidate a session with this option set to `false`?

**Comment by yotammadem** [comment]
1. Done
2. Maybe the name was misleading - this flag is not preventing you from setting session to empty. It is just preventing from empty sessions from being created in the first place. If you create a non empty session and then you make it empty it will not prevent you from doing this.
maybe we should call it **allowCreateEmptySession**. What do you think?

**Comment by yotammadem** [comment]
I just realized that I have to change the implementation.
I should check if there was no session on the incoming request, the session is empty and also the flag is false

**Comment by yotammadem** [comment]
Ok - now I pushed the fix:
When **allowEmptySession** is false:
1. It is not possible to create new empty session
2. It is possible to turn existing session to be empty
3. Non empty sessions are treated exactly like before adding this option

File: plugin.js [hunk 1]
```diff
@@ -29,12 +30,13 @@ function plugin (fastify, options, next) {
   }
 
   fastify.decorateRequest('session', {})
-
+  const isNewSession = Symbol('isNewSession')
```
Comment (jsumners) [inline plugin.js]: I think this can be defined at the same level as `defaultOptions`.

Comment (yotammadem) [reply plugin.js]: I tried this - the tests did not pass this way. I will try to figure out why

Comment (yotammadem) [reply plugin.js]: Ok, I figured it out
Now the symbol is at the top like you suggested

**Review by jsumners — APPROVED** [review]
LGTM

Thank you for the contribution.

**Pull request merged**